packages/graph/hash_graph/lib/graph/src/datastore/mod.rs

```diff
@@ -17,6 +17,119 @@ impl fmt::Display for DatastoreError {
     }
 }
 
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "clap", derive(clap::ArgEnum))]
+pub enum DatabaseType {
+    #[default]
+    Postgres,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "clap", derive(clap::Args))]
+pub struct DatabaseConnectionInfo {
+    /// The Database type to connect to
+    #[cfg_attr(feature = "clap", clap(long, default_value = "postgres", arg_enum))]
+    database_type: DatabaseType,
+
+    /// Database username
+    #[cfg_attr(
+        feature = "clap",
+        clap(long, default_value = "postgres", env = "HASH_GRAPH_USER")
+    )]
+    user: String,
+
+    /// Database password for authentication
+    #[cfg_attr(
+        feature = "clap",
+        clap(long, default_value = "postgres", env = "HASH_GRAPH_PASSWORD")
+    )]
+    password: String,
+
+    /// The host to connect to
+    #[cfg_attr(
+        feature = "clap",
+        clap(long, default_value = "localhost", env = "HASH_GRAPH_HOST")
+    )]
+    host: String,
+
+    /// The port to connect to
+    #[cfg_attr(
+        feature = "clap",
+        clap(long, default_value = "5432", env = "HASH_GRAPH_PORT")
+    )]
+    port: u16,
+
+    /// The database name to use
+    #[cfg_attr(
+        feature = "clap",
+        clap(long, default_value = "graph", env = "HASH_GRAPH_DATABASE")
+    )]
+    database: String,
+}
+
+impl DatabaseConnectionInfo {
+    pub const fn new(
+        database_type: DatabaseType,
+        user: String,
+        password: String,
+        host: String,
+        port: u16,
+        database: String,
+    ) -> Self {
+        Self {
+            database_type,
+            user,
+            password,
+            host,
+            port,
+            database,
+        }
+    }
+
+    fn url(&self) -> String {
```

Comment:
Do we perhaps want a docstring saying this is used as a connection URL but shouldn't be used for printing as it includes the password or something. (just a warning)

Or do we think the fact that we've implemented `Display` below is enough